### DISCUSSION TITLE
Expo router lazy loading

### DISCUSSION THREAD

**Discussion by m3tarayz** [expo/router#683] [Q&A]
hey guys!

I'm using `expo-router` to create a bottom tab navigation. The problem I'm currently facing is that some routes are being switched to quicker than others depending on the complexity/data loaded. It's as if the router waits until the whole content of the route is ready before switching to that route. Is there a way to enable lazy loading for expo router? I'm using quite a standard implementation.

  ```  
import { Navigator, Slot } from "expo-router";
import { TabRouter } from "@react-navigation/native";

<Navigator router={TabRouter}>
      <Slot />
</Navigator>   
```

I would like the loading part to be handled by the individual route after switching to it.
any help with this is appreciated, thanks!

**Comment by geronimosuarez** [comment]
you can try [Async Routes](https://docs.expo.dev/router/reference/async-routes/) but it'll apply lazy loading to all routes and it's still an experimental feature
